ampd/src/health_check.rs

```diff
@@ -0,0 +1,105 @@
+use error_stack::{Result, ResultExt};
+use std::net::SocketAddr;
+use thiserror::Error;
+use tracing::info;
+
+use axum::{http::StatusCode, routing::get, Json, Router};
+use serde::{Deserialize, Serialize};
+use tokio_util::sync::CancellationToken;
+
+pub struct Server {
+    listener: tokio::net::TcpListener,
+}
+
+#[derive(Error, Debug)]
+pub enum HealthCheckError {
+    #[error("Health check server error: {0}")]
+    Error(String),
+}
```

Comment:
Instead of using a single `Error(String)` variant to represent all errors, it would be better to define dedicated error variants for each possible error case. This allows errors to be handled statically rather than requiring string parsing at runtime.

Reply:
I was also thinking on that, but then i realised rest of the code doesn't need to react to this errors, thus, representing all of them as a single opaque one. Maybe there are other cases i am not seeing, let me know.